subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultToolchainSpec.java

```diff
@@ -35,8 +35,8 @@ public class DefaultToolchainSpec implements ToolchainSpecInternal {
     @Inject
     public DefaultToolchainSpec(ObjectFactory factory) {
         this.languageVersion = factory.property(JavaLanguageVersion.class);
-        this.vendor = factory.property(JvmVendorSpec.class).convention(DefaultJvmVendorSpec.any());
-        this.implementation = factory.property(JvmImplementation.class).convention(JvmImplementation.VENDOR_SPECIFIC);
+        this.vendor = factory.property(JvmVendorSpec.class).convention(getConventionVendor());
```

Comment:
I don't understand why we need a convention on vendor and implementation. Why not deal with the fact that those can be unset and then replace `null` by the default implementation? Maybe this is a question for @ljacomet.

Reply:
I kept it the same way it was, in order to not introduce any unintended behavior changes. I also don't entirely understand the idea behind these conventions. Maybe we could drop them in 8.0 and deal with nulls?

Reply:
I very much prefer having something that works with defaults and does not accept `null` in its _configured_ state. But maybe that's not the pattern for `Property` types.